alphagov/paas-bootstrap
# [#154437339] Upgrade bosh director and bosh-aws-cpi

[#154437339 Upgrade BOSH director and AWS CPI](https://www.pivotaltracker.com/story/show/154437339)

What?
-----

We want to upgrade to the latest version of bosh and bosh-aws-cpi:
 - [v264.7.0](https://github.com/cloudfoundry/bosh/releases/tag/v264.7.0)
 - [v69](https://github.com/cloudfoundry-incubator/bosh-aws-cpi-release/releases/tag/v69)

We are using 261.4 and 62

In this PR we must configure the mutual TLS for the NATs queue in bosh and all the clients: director, healthmanager, and agents:

 - We generate the certificates using the same approach than in paas-cf, instead of using the bosh v2 interpolation. We will migrate to bosh v2 interpolation in the future.
 - We must enable mutual TLS for bosh nats:
    - The certifiates must have CNs like: `*.client.bosh-internal`, `*.director.bosh-internal`, `*.hm.bosh-internal` or `*.nats.bosh-internal`.
    - The nats server cert must have a SAN for the host that the client use
 - We keep enabled the option [`nats.allow_legacy_agents:`](https://github.com/cloudfoundry/bosh/blob/master/jobs/nats/spec#L46) so the already non TLS agents can still work
 - We also generate a SSL cert for the director. Previously it was generated automatically.

Note: We had some trouble with the certificates due wrong values in the CN and SAN, see the commit messages.

How to review
---------

 * Code review
 * Push `make dev deployer-concourse pipelines DEPLOY_ENV=henry BRANCH=upgrade-bosh-and-cpi SELF_UPDATE_PIPELINE=false`
 * run the `create-bosh-concourse` pipeline:
   * It shall redeploy bosh with the new version
   * It shall redeploy concourse without errors
 * Redeploy CF: Shall redeploy without errors.
 * Check that all vms are working in `make bosh-cli`
 * Optionally: use `make ssh_bosh` and `make ssh_concourse` to check that the nats is using SSL
   * in bosh check `/var/vcap/jobs/nats/*`
   * in concourse check `/var/vcap/bosh/*`
   * check the logs.

Before merge
----------------

Update the last commit when the slow slow slow slow automated build in docker hub finishes https://hub.docker.com/r/governmentpaas/bosh-init/builds/b5e6yokk8yvgm5mjfagchid/

Who?
---

Anyone but @keymon or @henrytk
